### PR TITLE
refactor(shared): use new-style RxJS syntax

### PR DESCRIPTION
**Pull request by rohitkrai03**
- Use rxjs `pipe()` for chained operators.
- `Observable.of()` -> `of()`

### CODE REVIEW

File: src/app/shared/runtime-console/pipelines.service.ts [hunk 1]
```diff
@@ -3,6 +3,8 @@ import { Injectable } from '@angular/core';
 import { orderBy, take } from 'lodash';
 import { Context, Contexts } from 'ngx-fabric8-wit';
 import { Observable } from 'rxjs';
+import { combineLatest } from 'rxjs/observable/combineLatest';
```
Comment (joshuawilson) [inline src/app/shared/runtime-console/pipelines.service.ts]: Please convert:

```
import { combineLatest } from 'rxjs/observable/combineLatest';
a$.pipe(combineLatest(b$, c$));
```

into:

```
import { combineLatest } from 'rxjs';
combineLatest(a$, b$, c$);
```

File: src/app/profile/my-spaces/my-spaces-search-dialog/my-spaces-search-spaces-dialog.component.ts [hunk 2]
```diff
@@ -8,10 +8,11 @@ import {
 
 import {
   BehaviorSubject,
-  Observable,
   Subject,
   Subscription
 } from 'rxjs';
+import { combineLatest } from 'rxjs/observable/combineLatest';
```
Comment (joshuawilson) [inline src/app/profile/my-spaces/my-spaces-search-dialog/my-spaces-search-spaces-dialog.component.ts]: Please convert:

```
import { combineLatest } from 'rxjs/observable/combineLatest';
a$.pipe(combineLatest(b$, c$));
```

into:

```
import { combineLatest } from 'rxjs';
combineLatest(a$, b$, c$);
```

File: src/app/shared/context-current-user-guard.service.ts [hunk 3]
```diff
@@ -11,6 +11,8 @@ import { Notification, Notifications, NotificationType } from 'ngx-base';
 import { Contexts } from 'ngx-fabric8-wit';
 import { UserService } from 'ngx-login-client';
 import { Observable } from 'rxjs/Observable';
+import { combineLatest } from 'rxjs/observable/combineLatest';
```
Comment (joshuawilson) [inline src/app/shared/context-current-user-guard.service.ts]: Please convert:

```
import { combineLatest } from 'rxjs/observable/combineLatest';
a$.pipe(combineLatest(b$, c$));
```

into:

```
import { combineLatest } from 'rxjs';
combineLatest(a$, b$, c$);
```

File: src/app/profile/cleanup/cleanup.component.ts [hunk 4]
```diff
@@ -9,6 +9,10 @@ import { EventService } from '../../shared/event.service';
 import { IModalHost } from '../../space/wizard/models/modal-host';
 import { TenantService } from '../services/tenant.service';
 
+import { forkJoin } from 'rxjs/observable/forkJoin';
```
Comment (joshuawilson) [inline src/app/profile/cleanup/cleanup.component.ts]: This should just be 
```
import { forkJoin, of } from 'rxjs';
```

File: src/app/profile/my-spaces/my-spaces-search-dialog/my-spaces-search-spaces-dialog.component.ts (hunk 2 above)
Comment (rohitkrai03) [reply src/app/profile/my-spaces/my-spaces-search-dialog/my-spaces-search-spaces-dialog.component.ts]: `combineLatest` is not in `rxjs` for version 5.5.2 I think.

File: src/app/shared/context.service.ts [hunk 5]
```diff
@@ -14,6 +14,18 @@ import { FeatureTogglesService } from 'ngx-feature-flag';
 import { User, UserService } from 'ngx-login-client';
 import { Observable, Scheduler } from 'rxjs';
 import { ConnectableObservable } from 'rxjs/observable/ConnectableObservable';
+import { forkJoin } from 'rxjs/observable/forkJoin';
+import { merge } from 'rxjs/observable/merge';
+import { of } from 'rxjs/observable/of';
```
Comment (joshuawilson) [inline src/app/shared/context.service.ts]: This should just be 
```
import { ConnectableObservable, forkJoin, merge, Observable, of, Scheduler } from 'rxjs';
```

File: src/app/shared/runtime-console/space-namespace.service.ts [hunk 6]
```diff
@@ -7,6 +7,8 @@ import { Notifications, NotificationType } from 'ngx-base';
 import { Space, Spaces } from 'ngx-fabric8-wit';
 import { UserService } from 'ngx-login-client';
 import { Observable } from 'rxjs';
+import { forkJoin } from 'rxjs/observable/forkJoin';
```
Comment (joshuawilson) [inline src/app/shared/runtime-console/space-namespace.service.ts]: This should just be 
```
import { forkJoin, Observable } from 'rxjs';
```

File: src/app/shared/spaces.service.ts [hunk 7]
```diff
@@ -2,6 +2,10 @@ import { Injectable } from '@angular/core';
 import { Broadcaster } from 'ngx-base';
 import { Contexts, Space, Spaces, SpaceService } from 'ngx-fabric8-wit';
 import { Observable, ReplaySubject, Subject, Subscription } from 'rxjs';
+import { forkJoin } from 'rxjs/observable/forkJoin';
+import { merge } from 'rxjs/observable/merge';
+import { of } from 'rxjs/observable/of';
```
Comment (joshuawilson) [inline src/app/shared/spaces.service.ts]: This should just be 
```
import { forkJoin, of, merge } from 'rxjs';
```
(added to the main import of rxjs)

File: src/app/profile/my-spaces/my-spaces.component.spec.ts [hunk 8]
```diff
@@ -9,6 +9,7 @@ import { Context, Contexts, Space, SpaceService } from 'ngx-fabric8-wit';
 import { AuthenticationService, User, UserService } from 'ngx-login-client';
 import { Action } from 'patternfly-ng/action';
 import { Observable } from 'rxjs/Observable';
+import { of } from 'rxjs/observable/of';
```
Comment (joshuawilson) [inline src/app/profile/my-spaces/my-spaces.component.spec.ts]: This should just be 
```
import { Observable, of } from 'rxjs';
```

File: src/app/profile/overview/overview.component.spec.ts [hunk 9]
```diff
@@ -8,7 +8,7 @@ import { Notifications } from 'ngx-base/src/app/notifications/notifications';
 import { Contexts } from 'ngx-fabric8-wit';
 import { SpaceService } from 'ngx-fabric8-wit';
 import { UserService } from 'ngx-login-client';
-import { Observable } from 'rxjs/Observable';
+import { of } from 'rxjs/observable/of';
```
Comment (joshuawilson) [inline src/app/profile/overview/overview.component.spec.ts]: This should just be 
```
import { of } from 'rxjs';
```

File: src/app/profile/overview/spaces/overview-spaces.component.spec.ts [hunk 10]
```diff
@@ -4,6 +4,7 @@ import { ComponentFixture, TestBed } from '@angular/core/testing';
 import { Broadcaster, Logger } from 'ngx-base';
 import { Contexts, Fabric8WitModule, SpaceService, WIT_API_URL } from 'ngx-fabric8-wit';
 import { Observable } from 'rxjs/Observable';
+import { of } from 'rxjs/observable/of';
```
Comment (joshuawilson) [inline src/app/profile/overview/spaces/overview-spaces.component.spec.ts]: This should just be 
```
import { Observable, of } from 'rxjs';
```

File: src/app/profile/overview/work-items/work-items.component.spec.ts [hunk 11]
```diff
@@ -3,6 +3,7 @@ import { By } from '@angular/platform-browser';
 
 import { List, take } from 'lodash';
 import { Observable } from 'rxjs';
+import { of } from 'rxjs/observable/of';
```
Comment (joshuawilson) [inline src/app/profile/overview/work-items/work-items.component.spec.ts]: This should just be 
```
import { Observable, of } from 'rxjs';
```

File: src/app/profile/settings/connected-accounts/connected-accounts.spec.ts [hunk 12]
```diff
@@ -3,7 +3,8 @@ import { Component } from '@angular/core';
 import { Contexts } from 'ngx-fabric8-wit';
 import { AuthenticationService } from 'ngx-login-client';
 import { UserService } from 'ngx-login-client';
-import { Observable } from 'rxjs/Observable';
+import { Observable } from 'rxjs';
+import { of } from 'rxjs/observable/of';
```
Comment (joshuawilson) [inline src/app/profile/settings/connected-accounts/connected-accounts.spec.ts]: This should just be 
```
import { Observable, of } from 'rxjs';
```

File: src/app/profile/settings/feature-opt-in/feature-opt-in.component.spec.ts [hunk 13]
```diff
@@ -7,7 +7,7 @@ import { Notifications, NotificationType } from 'ngx-base';
 import { Feature, FeatureTogglesService } from 'ngx-feature-flag';
 import { UserService } from 'ngx-login-client';
 import { ListModule } from 'patternfly-ng';
-import { Observable } from 'rxjs';
+import { of } from 'rxjs/observable/of';
```
Comment (joshuawilson) [inline src/app/profile/settings/feature-opt-in/feature-opt-in.component.spec.ts]: This should just be 
```
import { of } from 'rxjs';
```

File: src/app/profile/spaces/spaces.component.spec.ts [hunk 14]
```diff
@@ -9,6 +9,7 @@ import { FeatureTogglesService } from 'ngx-feature-flag';
 import { AuthenticationService } from 'ngx-login-client';
 import { Subject } from 'rxjs';
 import { Observable } from 'rxjs/Observable';
+import { of } from 'rxjs/observable/of';
```
Comment (joshuawilson) [inline src/app/profile/spaces/spaces.component.spec.ts]: This should just be 
```
import { Observable, of } from 'rxjs';
```

File: src/app/profile/tenant/tenant.component.spec.ts [hunk 15]
```diff
@@ -9,6 +9,7 @@ import { Notifications } from 'ngx-base/src/app/notifications/notifications';
 import { Contexts, WIT_API_URL } from 'ngx-fabric8-wit';
 import { AuthenticationService, User, UserService } from 'ngx-login-client';
 import { Observable } from 'rxjs/Observable';
+import { of } from 'rxjs/observable/of';
```
Comment (joshuawilson) [inline src/app/profile/tenant/tenant.component.spec.ts]: This should just be 
```
import { Observable, of } from 'rxjs';
```

File: src/app/profile/update/update.component.spec.ts [hunk 16]
```diff
@@ -9,6 +9,7 @@ import { Notifications } from 'ngx-base/src/app/notifications/notifications';
 import { Contexts, WIT_API_URL } from 'ngx-fabric8-wit';
 import { AuthenticationService, User, UserService } from 'ngx-login-client';
 import { Observable } from 'rxjs/Observable';
+import { of } from 'rxjs/observable/of';
```
Comment (joshuawilson) [inline src/app/profile/update/update.component.spec.ts]: This should just be 
```
import { Observable, of } from 'rxjs';
```

File: src/app/shared/analytics.service.spec.ts [hunk 17]
```diff
@@ -3,7 +3,8 @@ import { Router } from '@angular/router';
 import { Broadcaster } from 'ngx-base';
 import { Contexts, Spaces } from 'ngx-fabric8-wit';
 import { UserService } from 'ngx-login-client';
-import { Observable, Subject } from 'rxjs';
+import { Subject } from 'rxjs';
+import { of } from 'rxjs/observable/of';
```
Comment (joshuawilson) [inline src/app/shared/analytics.service.spec.ts]: This should just be 
```
import { of, Subject } from 'rxjs';
```

File: src/app/shared/auth-guard.service.ts [hunk 18]
```diff
@@ -10,6 +10,7 @@ import {
 import { Logger } from 'ngx-base';
 import { AuthenticationService } from 'ngx-login-client';
 import { Observable } from 'rxjs';
+import { of } from 'rxjs/observable/of';
```
Comment (joshuawilson) [inline src/app/shared/auth-guard.service.ts]: This should just be 
```
import { Observable, of } from 'rxjs';
```

File: src/app/shared/context-resolver.service.ts [hunk 19]
```diff
@@ -7,6 +7,8 @@ import {
 
 import { Context } from 'ngx-fabric8-wit';
 import { Observable } from 'rxjs';
+import { of } from 'rxjs/observable/of';
```
Comment (joshuawilson) [inline src/app/shared/context-resolver.service.ts]: This should just be 
```
import { Observable, of } from 'rxjs';
```

File: src/app/shared/context.service.spec.ts [hunk 20]
```diff
@@ -7,7 +7,7 @@ import { Broadcaster, Notifications } from 'ngx-base';
 import { SpaceService } from 'ngx-fabric8-wit';
 import { FeatureTogglesService } from 'ngx-feature-flag';
 import { UserService } from 'ngx-login-client';
-import { Observable } from 'rxjs';
+import { of } from 'rxjs/observable/of';
```
Comment (joshuawilson) [inline src/app/shared/context.service.spec.ts]: This should just be 
```
import { of } from 'rxjs';
```

File: src/app/shared/interceptors/cache.interceptor.spec.ts [hunk 21]
```diff
@@ -14,6 +14,7 @@ import {
 } from '@angular/common/http/testing';
 import { fakeAsync, TestBed, tick } from '@angular/core/testing';
 import { Observable } from 'rxjs';
+import { of } from 'rxjs/observable/of';
```
Comment (joshuawilson) [inline src/app/shared/interceptors/cache.interceptor.spec.ts]: This should just be 
```
import { Observable, of } from 'rxjs';
```

File: src/app/shared/login.service.spec.ts [hunk 22]
```diff
@@ -3,6 +3,7 @@ import { Router } from '@angular/router';
 import { RouterTestingModule } from '@angular/router/testing';
 import { LocalStorageService } from 'angular-2-local-storage';
 import { Observable } from 'rxjs';
+import { of } from 'rxjs/observable/of';
```
Comment (joshuawilson) [inline src/app/shared/login.service.spec.ts]: This should just be 
```
import { Observable, of } from 'rxjs';
```

File: src/app/shared/login.service.ts [hunk 23]
```diff
@@ -4,7 +4,8 @@ import { Router } from '@angular/router';
 import { LocalStorageService } from 'angular-2-local-storage';
 import { Broadcaster, Notification, Notifications, NotificationType } from 'ngx-base';
 import { AUTH_API_URL, AuthenticationService, UserService } from 'ngx-login-client';
-import { Observable } from 'rxjs';
+import { of } from 'rxjs/observable/of';
```
Comment (joshuawilson) [inline src/app/shared/login.service.ts]: This should just be 
```
import { of } from 'rxjs';
```

File: src/app/shared/profile-resolver.service.ts [hunk 24]
```diff
@@ -9,6 +9,8 @@ import {
 import { Context } from 'ngx-fabric8-wit';
 import { UserService } from 'ngx-login-client';
 import { Observable } from 'rxjs';
+import { of } from 'rxjs/observable/of';
```
Comment (joshuawilson) [inline src/app/shared/profile-resolver.service.ts]: This should just be 
```
import { Observable, of } from 'rxjs';
```

File: src/app/shared/space-template.service.ts [hunk 25]
```diff
@@ -4,6 +4,7 @@ import { HttpClient } from '@angular/common/http';
 
 import { WIT_API_URL } from 'ngx-fabric8-wit';
 import { Observable } from 'rxjs/Observable';
+import { map } from 'rxjs/operators';
```
Comment (joshuawilson) [inline src/app/shared/space-template.service.ts]: This should just be 
```
import { Observable } from 'rxjs';
```

File: src/app/shared/spaces.service.spec.ts [hunk 26]
```diff
@@ -4,6 +4,7 @@ import { Broadcaster } from 'ngx-base';
 import { Context, Contexts, Space, SpaceService } from 'ngx-fabric8-wit';
 import { User } from 'ngx-login-client';
 import { ConnectableObservable, Observable } from 'rxjs';
+import { of } from 'rxjs/observable/of';
```
Comment (joshuawilson) [inline src/app/shared/spaces.service.spec.ts]: This should just be 
```
import { ConnectableObservable, Observable, of } from 'rxjs';
```

File: src/app/profile/cleanup/cleanup.component.spec.ts [hunk 27]
```diff
@@ -6,12 +6,13 @@ import { Router } from '@angular/router';
...
 import { Contexts, SpaceService, WIT_API_URL } from 'ngx-fabric8-wit';
 import { AuthenticationService, UserService } from 'ngx-login-client';
-import { Observable } from 'rxjs/Observable';
-
 import { EventService } from '../../shared/event.service';
 import { TenantService } from '../services/tenant.service';
 import { CleanupComponent } from './cleanup.component';
 
+import { Observable } from 'rxjs';
+import { of } from 'rxjs/observable/of';
```
Comment (joshuawilson) [inline src/app/profile/cleanup/cleanup.component.spec.ts]: This should just be 
```
import { Observable, of } from 'rxjs';
```

**Pull request merged**